src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -199,19 +289,21 @@ private bool IsValidProjectFilePath(string path)
             return globalProperties;
         }
 
-        /// <param name="propertyToDisableIfTrue">A property that will be disabled (NOT BY THIS FUCTION) based on the condition.</param>
-        /// <returns>Returns true if Configuration on a project is not Debug or Release. Note if someone explicitly set Debug, we don't detect that here.</returns>
-        /// <remarks>Will log that the property will not work if this is true.</remarks>
-        private bool ProjectHasUserCustomizedConfiguration(ProjectInstance project, string propertyToDisableIfTrue)
+        /// <summary>
+        /// Because command-line options that translate to MSBuild properties aren't in the global arguments from Properties, we need to add the TargetFramework to the collection.
+        /// The TargetFramework is the only command-line option besides Configuration that could affect the pre-evaluation.
+        /// This allows the pre-evaluation to correctly deduce its Publish or PackRelease value because it will know the actual TargetFramework being used.
+        /// </summary>
+        /// <param name="oldGlobalProperties">The set of MSBuild properties that were specified explicitly like -p:Property=Foo or in other syntax sugars.</param>
+        /// <returns>The same set of global properties for the project, but with the new potential TFM based on -f or --framework.</returns>
+        void InjectTargetFrameworkIntoGlobalProperties(Dictionary<string, string> oldGlobalProperties)
```

Comment:
Name change suggestion:

```suggestion
        void InjectTargetFrameworkIntoGlobalProperties(Dictionary<string, string> globalProperties)
```

Reply:
Update parameter name to reflect the pass by reference default of C#